edu.gemini.lch.model/src/main/java/edu/gemini/lch/model/Interval.scala

```diff
@@ -0,0 +1,28 @@
+package edu.gemini.lch.model
+
+import java.time.Instant
+
+/**
+ * An interval is a period of time between two instants.
+ * Includes start, excludes end.
+ * Requirement: end >= start.
+ * They have fixed ms duration, i.e. end - start and are not comparable.
+ * To compare intervals, you should compare their lengths, i.e. Durations.
+ */
+case class Interval(start: Instant, end: Instant) {
+  def overlaps(other: Interval): Boolean = {
+    val thisStart = start.toEpochMilli
+    val thisEnd   = end.toEpochMilli
+
+    // If null, assume now and check if now falls in the represented interval.
+    if (other == null) {
+      val now: Instant = Instant.now()
+      thisStart < now.toEpochMilli && now.toEpochMilli < thisEnd
+    }
+    else {
+      val otherStart = other.start.toEpochMilli
+      val otherEnd   = other.end.toEpochMilli
+      thisStart < otherEnd && otherStart < thisEnd
```

Comment:
You can compare `Instant` directly if you import `scala.math.Ordering.Implicits._`.  As in:

```
> import java.time.Instant 
import java.time.Instant

> import scala.math.Ordering.Implicits._ 
import scala.math.Ordering.Implicits._

> val t0 = Instant.now 
t0: Instant = 2019-11-04T13:36:18.852Z

> val t1 = t0.plusMillis(1L) 
t1: Instant = 2019-11-04T13:36:18.853Z

> t0 < t1 
res4: Boolean = true

> t1 < t0 
res5: Boolean = false
```

Reply:
Excellent. This will likely apply to a few places.